sidekiq_alive.gemspec

```
@@ -50,7 +50,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency("rubocop-shopify", "~> 2.10")
   spec.add_development_dependency("solargraph", "~> 0.49.0")
 
-  spec.add_dependency("rack", "< 3")
+  spec.add_dependency("rack", "< 4")
```

If the small subset of rack API that we use did not change we can add support this way:
```suggestion
  spec.add_dependency("rack", ">= 2", "<= 4")
```

---

The tests are not passing, the only way to support this is to modify the code for each version

---

CC: @andrcuns 👆

---

>If the small subset of rack API that we use did not change we can add support this way

That's the issue, we can't. Not only the API changed, `rackup` has been moved to a separate gem entirely starting from version 3. So it's not as straight forward as with sidekiq. We need to have `rackup` gem as runtime dependency and `rackup` can only be used with `rack >= 3`. That's why I'm not sure how to support both.

---

Rack is the only player in town and all the web servers in ruby depend on it.
https://www.reddit.com/r/ruby/comments/ka965h/why_is_rack_the_only_game_in_town_for_ruby/
I do not feel like implement myself TCP/IP protocol parser.
The only solution I find is to remove the dependency and add a rust server implementation 🤣 .